tests/test_references.py

```diff
@@ -419,3 +419,16 @@ def test_reference_wrong_type(self):
         with pytest.raises(DataGenError) as e:
             generate(StringIO(yaml))
         assert "incorrect object type" in str(e).lower()
+
+    def test_reference_really_wrong_type(self, disable_typeguard):
+        yaml = """
+        - object: B
+          count: 2
+          fields:
+              A_ref:
+                reference: 5"""
+        with mock.patch(
+            "typeguard.check_argument_types", lambda *args, **kwargs: ...
+        ), pytest.raises(DataGenError) as e:
```

Comment:
It looks to me like it would actually be appropriate to type the `reference` function as accepting Any, because it does have logic for handling any type.

Also you're patching typeguard twice

Reply:
Okay, I switched to Any to make it easier to switch to MyPy some day.